app.py

```diff
@@ -0,0 +1,537 @@
+        'total': float(request.form.get('total')),
+        'car': {
+            'make': request.form.get('make'),
+            'model': request.form.get('model'),
+            'trim': request.form.get('trim'),
+            'year': request.form.get('year'),
+            'price': float(request.form.get('price'))
+        },
+        'votes':{
+            'like': {
+                'count': 1,
+                'users_liked': users_Array
+            },
+            'dislike': {
+                'count': 0,
+                'users_disliked': []
+            }
+        },
+    }
+
+   # Adds exterior collection to record 
+    exterior_dict = {}
+    for item in exterior:
+        exterior_dict.update({
+            item["part_id"]: {
+            'product': request.form.get('exterior_'+item["part_id"]+'_product'),
+            'link': request.form.get('exterior_'+item["part_id"]+'_link'),
+            'price': float(request.form.get('exterior_'+item["part_id"]+'_price'))
+            }
+        })
+
+    record.update({'exterior': exterior_dict})
+        
+    # Adds Engine collection to record 
+    engine_dict = {}
+    for item in engine:
+        engine_dict.update({
+            item["part_id"]: {
+            'product': request.form.get('engine_'+item["part_id"]+'_product'),
+            'link': request.form.get('engine_'+item["part_id"]+'_link'),
+            'price': float(request.form.get('engine_'+item["part_id"]+'_price'))
+            }
+        })
+
+    record.update({'engine': engine_dict})
+
+    # Adds Running Gear collection to record 
+    running_dict = {}
+    for item in running:
+        running_dict.update({
+            item["part_id"]: {
+            'product': request.form.get('running_'+item["part_id"]+'_product'),
+            'link': request.form.get('running_'+item["part_id"]+'_link'),
+            'price': float(request.form.get('running_'+item["part_id"]+'_price'))
+            }
+        })
+
+    record.update({'running': running_dict})
+
+    # Adds Interior collection to record 
+    interior_dict = {}
+    for item in interior:
+        interior_dict.update({
+            item["part_id"]: {
+            'product': request.form.get('interior_'+item["part_id"]+'_product'),
+            'link': request.form.get('interior_'+item["part_id"]+'_link'),
+            'price': float(request.form.get('interior_'+item["part_id"]+'_price'))
+            }
+        })
+
+    record.update({'interior': interior_dict})
+    builds.insert_one(record)
+    flash('Build Created', category='success')
+    return redirect(url_for('builds'))
+
+# View a Record
+@app.route("/view/<build_id>")
+def view_record(build_id):
+    build = mongo.db.builds.find_one({
+        "_id": ObjectId(build_id)
+    })
+
+    builds = mongo.db.builds
+    exterior = mongo.db.exterior.find()
+    engine = mongo.db.engine.find()
+    running = mongo.db.runninggear.find()
+    interior = mongo.db.interior.find()
```

Comment:
you shouldn't have to do all this in this route, think of a better way to achieve what you want to